File: flutter_app/lib/models/notification_model.dart
class NotificationModel {
  final String id;
  final String title;
  final String body;
  final String? type;
  final Map<String, dynamic>? data;
  final bool isRead;
  final DateTime createdAt;

  NotificationModel({
    required this.id,
    required this.title,
    required this.body,
    this.type,
    this.data,
    required this.isRead,
    required this.createdAt,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['id'],
      title: json['title'],
      body: json['body'],
      type: json['type'],
      data: json['data'] is Map ? json['data'] : null,
      isRead: json['is_read'] ?? json['isRead'] ?? false,
      createdAt: DateTime.parse(json['created_at'] ?? json['createdAt']),
    );
  }
}

File: backend/src/models/notification.model.js
import { query } from '../config/database.js';

export const createNotification = async (userId, title, body, type, data = {}) => {
  const result = await query(`
    INSERT INTO notifications (user_id, title, body, type, data)
    VALUES ($1, $2, $3, $4, $5)
    RETURNING *
  `, [userId, title, body, type, JSON.stringify(data)]);
  return result.rows[0];
};

export const getUserNotifications = async (userId, limit = 20, offset = 0) => {
  const result = await query(`
    SELECT * FROM notifications
    WHERE user_id = $1
    ORDER BY created_at DESC
    LIMIT $2 OFFSET $3
  `, [userId, limit, offset]);
  return result.rows;
};

export const markAsRead = async (id, userId) => {
  await query(`
    UPDATE notifications SET is_read = true 
    WHERE id = $1 AND user_id = $2
  `, [id, userId]);
};

export const markAllAsRead = async (userId) => {
  await query(`
    UPDATE notifications SET is_read = true 
    WHERE user_id = $1
  `, [userId]);
};
